pruner/graph.go

```diff
@@ -0,0 +1,328 @@
+package pruner
```

Comment:
Let's put this into `registry/storage/pruner` to be clear it is tied to storage backends.